src/Swashbuckle.AspNetCore.Annotations/AnnotationsOperationFilter.cs

```diff
@@ -13,13 +13,13 @@ public void Apply(Operation operation, OperationFilterContext context)
         {
             if (context.MethodInfo == null) return;
 
+            var controllerAttributes = context.MethodInfo.DeclaringType.GetTypeInfo().GetCustomAttributes(true).Reverse(); // Reverse so most derived types are last
```

Comment:
It would add some complexity, but I guess the proper fix for this would be to walk the class hierarchy of the declaring types of the attributes of the same types and work out which one is the winner that way. Do you want me to add something like that?

Reply:
It's certainly going to add a ton of complexity. I would suggest we remove the `Reverse` calls above and just accept that _this_ particular PR only solves the "controller vs action" precedence problem. Then I can close this PR and at least get that fix out. If you really need the direct vs inherited precedence to hold as well, then let's open a second PR  and tackle that separately?

Reply:
BTW - I wasn't able to find any documentation that states whether or not `GetCustomAttributes` preserves any kind of ordering. You may want to dig a little deeper though - if that's stated on any reputable source, then I'd be happy to go with the `Reverse` approach that you've applied above.

Reply:
Yeah I had a quick skim, but couldn't find anything that guaranteed the ordering returned. I'll see if I can dig through a bit more and find something to save the need to do the class hierarchy walk.

Happy to do that separately in a different PR (today or tomorrow), so I can drop the `Reverse()`. Would you rather I remove the test that will fail when I do that, or can I just mark it as skip for now for when I do the follow-up PR?

Reply:
I think remove the test for now, and re-introduce with the follow-up PR

Reply:
Changed in 446e83ceabcdbb88ea73a8907e4a60eb13501a0a.